### PR TITLE
[feature](meta) Add thrift rpc to get db/table/backend meta

### PR DESCRIPTION
**Pull request by JackDrogon**
## Proposed changes

Add thrift get db/table/backend meta

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by JackDrogon** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 1]
```diff
@@ -3331,4 +3338,174 @@ public TCreatePartitionResult createPartition(TCreatePartitionRequest request) t
...
+            result = getMetaImpl(request, clientAddr);
+        } catch (UserException e) {
+            LOG.warn("failed to get meta: {}", e.getMessage());
+            status.setStatusCode(TStatusCode.ANALYSIS_ERROR);
+            status.addToErrorMsgs(e.getMessage());
+        } catch (Throwable e) {
+            LOG.warn("catch unknown result.", e);
+            status.setStatusCode(TStatusCode.INTERNAL_ERROR);
+            status.addToErrorMsgs(Strings.nullToEmpty(e.getMessage()));
+            return result;
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:3359]:
```suggestion
```

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 2]
```diff
@@ -3331,4 +3338,174 @@ public TCreatePartitionResult createPartition(TCreatePartitionRequest request) t
...
+            }
+        }
+
+        // Step 3: get meta
+        try {
+            result = Env.getMeta(db, tables);
+            return result;
+        } catch (Throwable e) {
+            throw e;
+        }
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:3441]: Just `return Env.getMeta(db, tables);`   ?

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 3]
```diff
@@ -3331,4 +3338,174 @@ public TCreatePartitionResult createPartition(TCreatePartitionRequest request) t
...
+            result = getBackendMetaImpl(request, clientAddr);
+        } catch (UserException e) {
+            LOG.warn("failed to get backend meta: {}", e.getMessage());
+            status.setStatusCode(TStatusCode.ANALYSIS_ERROR);
+            status.addToErrorMsgs(e.getMessage());
+        } catch (Throwable e) {
+            LOG.warn("catch unknown result.", e);
+            status.setStatusCode(TStatusCode.INTERNAL_ERROR);
+            status.addToErrorMsgs(Strings.nullToEmpty(e.getMessage()));
+            return result;
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:3461]:
```suggestion
```

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 4]
```diff
@@ -5614,6 +5623,89 @@ public static boolean isTableNamesCaseInsensitive() {
...
+                partitionMeta.setId(partitionId);
+                partitionMeta.setName(partition.getName());
+                String partitionRange = "";
+                if (tblPartitionInfo.getType() == PartitionType.RANGE
+                        || tblPartitionInfo.getType() == PartitionType.LIST) {
+                    partitionRange = tblPartitionInfo.getItem(partitionId).getItems().toString();
+                }
+                partitionMeta.setRange(partitionRange);
+                partitionMeta.setVisibleVersion(partition.getVisibleVersion());
+                // partitionMeta.setTemp（partition.isTemp());
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:5650]: Isn't it needed here anymore?

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 5]
```diff
@@ -3331,4 +3338,174 @@ public TCreatePartitionResult createPartition(TCreatePartitionRequest request) t
...
+
+            if (getMetaDb.isSetId()) {
+                db = Env.getCurrentInternalCatalog().getDbNullable(getMetaDb.getId());
+            } else if (getMetaDb.isSetName()) {
+                db = Env.getCurrentInternalCatalog().getDbNullable(getMetaDb.getName());
+            }
+
+            if (db == null) {
+                LOG.warn("db not found {}", getMetaDb);
+                return result;
```
Comment (w41ter) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:3398]: Is returns OK if db does not exist by design?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java (hunk 4 above)
Comment (JackDrogon) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:5650]: mark for next add temp partition support

Comment (JackDrogon) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:5650]: It's just a comment, need to use another difficult way to judge temp in doris

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java (hunk 1 above)
Comment (JackDrogon) [reply fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java]: done

**Comment by JackDrogon** [comment]
run buildall

**Review by w41ter — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**